vioserial/sys/Control.c

```diff
@@ -43,23 +43,18 @@ VIOSerialSendCtrlMsg(
     sg.physAddr = MmGetPhysicalAddress(&cpkt);
     sg.length = sizeof(cpkt);
 
-    WdfSpinLockAcquire(pContext->CVqLock);
+    WdfWaitLockAcquire(pContext->COutVqLock, NULL);
     if(0 <= virtqueue_add_buf(vq, &sg, 1, 0, &cpkt, NULL, 0))
     {
         virtqueue_kick(vq);
         while(!virtqueue_get_buf(vq, &len))
         {
            KeStallExecutionProcessor(50);
-           if(++cnt > RETRY_THRESHOLD)
-           {
-              TraceEvents(TRACE_LEVEL_FATAL, DBG_PNP, "<-> %s retries = %d\n", __FUNCTION__, cnt);
-              break;
-           }
         }
     }
-    WdfSpinLockRelease(pContext->CVqLock);
+    WdfWaitLockRelease(pContext->COutVqLock);
 
-    TraceEvents(TRACE_LEVEL_INFORMATION, DBG_PNP, "<-- %s cnt = %d\n", __FUNCTION__, cnt);
```

Comment:
nit: cnt is now unused

`int cnt = 0;`

at the top of the function can be deleted.

Reply:
My fault, forgot to delete